### ISSUE TITLE
[CI] EsExecutorsTests testRejectionMessageAndShuttingDownFlag test failing

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#52378]
This test has suddenly started failing in `master` with 8 failures already today. 

```
java.lang.AssertionError: 

Expected: a string containing "of dummy runnable"
     but: was "rejected execution of org.elasticsearch.common.util.concurrent.TimedRunnable@602a0ff8 on EWMATrackingEsThreadPoolExecutor[name = org.elasticsearch.common.util.concurrent.EsExecutorsTests/testRejectionMessageAndShuttingDownFlag, queue capacity = 42, task execution EWMA = 0s, org.elasticsearch.common.util.concurrent.EWMATrackingEsThreadPoolExecutor@18d8cffa[Running, pool size = 10, active threads = 10, queued tasks = 42, completed tasks = 0]]"
Open stacktrace
```

Couple of examples:
https://gradle-enterprise.elastic.co/s/mxn3kzfskwjgc/tests/nfxodyych4uas-o3edcuj7caqvm
https://gradle-enterprise.elastic.co/s/d2ogvhzu5xzdg/tests/nfxodyych4uas-o3edcuj7caqvm
https://gradle-enterprise.elastic.co/s/23htwmefsmcb4/tests/nfxodyych4uas-o3edcuj7caqvm

@ywelsch perhaps this is related to your recent changes?

**Comment by mark-vieira** [comment]
This has been muted in `master` with e4a399f68096b8e851869e8ffebcf1c242bdc587.

**Comment by ywelsch** [comment]
fix was merged here: 36436ff5b6c
Can you unmute (I'm off into the weekend)?

**Comment by mark-vieira** [comment]
> Can you unmute (I'm off into the weekend)?

Done. Thanks, Yannick!
